exchange/bitswap/bitswap.go

```diff
@@ -376,21 +361,24 @@ func (bs *Bitswap) PeerDisconnected(p peer.ID) {
 	bs.engine.PeerDisconnected(p)
 }
 
-func (bs *Bitswap) cancelBlocks(ctx context.Context, bkeys []u.Key) {
+func (bs *Bitswap) cancelBlocks(ctx context.Context, bkeys []u.Key) error {
 	if len(bkeys) < 1 {
-		return
+		return nil
 	}
 	message := bsmsg.New()
 	message.SetFull(false)
 	for _, k := range bkeys {
+		log.Debug("cancel block: %s", k)
 		message.Cancel(k)
 	}
 	for _, p := range bs.engine.Peers() {
 		err := bs.send(ctx, p, message)
```

Comment:
does send wait until it finishes? cause if so, it should probably happen in parallel for all peers.